test/unistd/access.c

```diff
@@ -65,14 +66,73 @@ int main() {
   printf("F_OK(%s): %d\n", "renamedfile", faccessat(AT_FDCWD, "renamedfile", F_OK, 0));
   printf("errno: %d\n", errno);
 
+  struct stat fileStats;
+  stat("fchmodtest", &fileStats);
+#if !defined(NODEFS) && !defined(NODERAWFS)
+  assert(fileStats.st_mode & 0666);
+#else
+  assert(fileStats.st_mode & 0644);
+#endif
+  
+  EM_ASM(
+    var fchmodstream = FS.open("fchmodtest", "r");
+#if WASMFS
+    FS.fchmod(fchmodstream, 0000);
+#else
+    FS.fchmod(fchmodstream.fd, 0000);
```

Comment:
Seeing the ifdef for `.fd` here makes me wonder if we should maybe just try to match the JS FS here. That is, we could return `{ fd: fd }` rather than just `fd`.

I'm not sure though. Perhaps we can discuss with @tlively when possible. But that doesn't need to block this PR, if we decide on changes later we can revisit this.